nrf52-hal-common/src/uarte.rs

```diff
@@ -0,0 +1,153 @@
+//! HAL interface to the UARTE peripheral
+//!
+//! See product specification, chapter 33.
+use core::ops::Deref;
+
+use target::{uarte0, UARTE0};
+
+use gpio::{p0::P0_Pin, Output, PushPull};
+use prelude::*;
+
+// Re-export SVD variants to allow user to directly set values
+pub use target::uarte0::baudrate::BAUDRATEW as Baudrate;
+pub use target::uarte0::config::PARITYW as Parity;
+
+pub trait UarteExt: Deref<Target = uarte0::RegisterBlock> + Sized {
+    fn constrain(self, pins: Pins, parity: Parity, baudrate: Baudrate) -> Uarte<Self>;
+}
+
+impl UarteExt for UARTE0 {
+    fn constrain(self, pins: Pins, parity: Parity, baudrate: Baudrate) -> Uarte<Self> {
+        Uarte::new(self, pins, parity, baudrate)
+    }
+}
+
+/// Interface to a UARTE instance
+///
+/// This is a very basic interface that comes with the following limitations:
+/// - The UARTE instances share the same address space with instances of UART.
+///   You need to make sure that conflicting instances
+///   are disabled before using `Uarte`. See product specification, section 15.2.
```

Comment:
That reference to the product specification is specific to nRF52832. It's section 6.1.2 for nRF52840.

Reply:
Ah, nice. I don't have a copy of the 840 OPS (haven't worked on that chip before), so I will make sure to pull that down and update it. I was hoping they would be laid out the same :)